### ISSUE TITLE
CS' java 8 code isn't compilable

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#2880]
While looking at our existing Java 8 code to try and work on new stuff, I noticed a few of our 'non-compilable' Java 8 files aren't actually compilable. Eclipse gives me around 100 errors. This can be seen in travis (https://travis-ci.org/rnveach/checkstyle/builds/107475572) when run with the command:
````
  include:
    # full java 8 tests (oraclejdk8)
    - jdk: oraclejdk8
      env: DESC="full java 8" CMD="sed -i 's/<tools\.jar\.version>1\.7/<tools\.jar\.version>1\.8/g' pom.xml && sed -i 's/<java\.version>1\.7/<java\.version>1\.8/g' pom.xml && cp -R src/test/resources-noncompilable/* src/test/resources && mvn clean install" COVERAGE_CMD="" DEPLOY="true"
````

We should add something like this command as a new travis node make these compilable so we are sure we are building a working parser.

**Comment by romani** [comment]
not all Inputs could be compilable, we need some Inputs as damaged on purpose.

but all java8 related hast to be compilable, we could reuse `javac` to compile them

**Comment by romani** [comment]
fixed, new issue is created #2904 .
